plugins/cryptokiplugin/plugin.cpp

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (C) 2017 Jolla Ltd.
+ * Contact: Chris Adams <chris.adams@jollamobile.com>
+ * All rights reserved.
+ * BSD 3-Clause License, see LICENSE *
+ *
+ * Copyright (C) 2017 Open Mobile Platform LLC.
+ * Contact: Denis Semakin <d.semakin@omprussia.ru>
+ * All rights reserved.
+ */
+
+#include "plugin.h"
+
+#include "Crypto/key.h"
+#include "Crypto/certificate.h"
+
+#include <QtCore/QByteArray>
+#include <QtCore/QMap>
+#include <QtCore/QVector>
+#include <QtCore/QString>
+#include <QtCore/QCryptographicHash>
+
+Q_PLUGIN_METADATA(IID Sailfish_Crypto_CryptoPlugin_IID)
+
+Sailfish::Crypto::Daemon::Plugins::CryptokiPlugin::CryptokiPlugin(QObject *parent)
+    : Sailfish::Crypto::CryptoPlugin(parent)
+{
+    //osslevp_init();
+    //TODO: WIP
+}
+
+Sailfish::Crypto::Daemon::Plugins::CryptokiPlugin::~CryptokiPlugin()
+{
+}
+
+QVector<Sailfish::Crypto::Key::Algorithm>
+Sailfish::Crypto::Daemon::Plugins::CryptokiPlugin::supportedAlgorithms() const
+{
+    QVector<Sailfish::Crypto::Key::Algorithm> retn;
+    retn.append(Sailfish::Crypto::Key::Aes256);
```

Comment:
API question: should we separate the algorithm from the keysize, or does it make sense to "combine" those, as it is currently in the prototype?